composer/utils/auto_log_hparams.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Contains helper functions for auto-logging hparams."""
+
+from typing import Any, Dict, List, Tuple
+
+__all__ = ['extract_hparams', 'convert_nested_dict_to_flat_dict', 'convert_flat_dict_to_nested_dict']
+
+
+def extract_hparams(locals_dict: Dict[str, Any]) -> Dict[str, Any]:
+    """Takes in local symbol table and recursively grabs any hyperparameter.
+
+    Args:
+        locals_dict (Dict[str, Any]): The local symbol table returned when calling locals(),
+            which maps any free local variables' names to their values.
+
+    Returns:
+        Dict[str, Any]: A nested dictionary with every element of locals_dict mapped to its
+            value or to another sub_dict.
+    """
+    hparams = {}
+    if 'self' in locals_dict:
+        locals_dict.pop('self')
+
+    for k, v in locals_dict.items():
```

Comment:
should we exclude the key `_`? It showed up for me when I ran it locally

Reply:
fixed